cmd/ftl/cmd_schema_diff.go

```diff
@@ -57,6 +83,60 @@ func (d *schemaDiffCmd) Run(ctx context.Context, currentURL *url.URL) error {
 	return nil
 }
 
+func localSchema(moduleDirs projectconfig.Config) (*schema.Schema, error) {
+	pb := &schemapb.Schema{}
+	found := false
+	tried := ""
+	moduleFound := false
+	moduleTried := ""
+	for _, modulePaths := range moduleDirs.AbsModuleDirs() {
```

Comment:
@alecthomas is this the correct way to do local discovery of available modules? Basically I am just scanning sub directories of the modules directory looking for `.ftl/schema.pb` files, but I have a feeling that is wrong (and certainly won't work for multi module maven projects).
Should I just do a full recursive scan of all directories under the roots for `.ftl/schema.pb`, or is there more complex logic elsewhere I should re-use?

Reply:
This is not quite correct unfortunately. `module-dirs` points at the "roots", not at the individual modules. That said, there should be some existing code in `buildengine` that already does what you want (in some form), so refactor that so that it's reusable.